contrib/src/main/scala/akka/stream/contrib/AccumulateWhileUnchanged.scala

```diff
@@ -61,21 +81,20 @@ final class AccumulateWhileUnchanged[Element, Property](propertyExtractor: Eleme
 
         if (currentState.isEmpty) currentState = Some(nextState)
 
-        currentState match {
-          case Some(`nextState`) =>
+        (currentState, maxElements) match {
+          case (Some(`nextState`), max) if max.isEmpty || nbElements < max.get =>
```

Comment:
I would like to avoid calling `.get`. It is an unsafe operation (can throw an exception). It is only safe to call it depending on the context. Here it is currently fine, since we check that `max` is not empty, but such code does not age well during refactorings.

How about extracting the following to a private method called for example `stash`:
```scala
buffer += nextElement
nbElements += 1
pull(in)
```

and then calling that from:
```scala
(currentState, maxElements) match {
  case (Some(`nextState`), None) => stash(nextElement)
  case (Some(`nextState`), Some(max)) if nbElements < max => stash(nextElement)
  case _ => pushResults(Some(nextElement), Some(nextState))
}
```

Reply:
Thank you for your detailed explanation!
I fixed it as suggested.